lib/blocks.php

```diff
@@ -191,6 +195,112 @@ function gutenberg_register_core_block_styles( $block_name ) {
 	}
 }
 
+/**
+ * Change the way block styles get loaded depending on their size.
+ *
+ * Optimizes performance and sustainability of block styles.
+ *
+ * @return void
+ */
+function gutenberg_optimize_block_styles_loading() {
+
+	$total_inline_limit = 20000;
+	/**
+	 * The maximum size of inlined styles in bytes.
+	 *
+	 * @param int $total_inline_limit The file-size threshold, in bytes.
+	 * @return int
+	 */
+	$total_inline_limit = apply_filters( 'styles_inline_size_limit', $total_inline_limit );
+
+	global $wp_styles;
+	$styles = array();
+
+	// Build an array of styles that have a path defined.
+	foreach ( $wp_styles->queue as $handle ) {
+		if (
+			isset( $wp_styles->registered[ $handle ] ) &&
```

Comment:
If it would be possible to set it as data, then you can also get it like here:

https://github.com/WordPress/wordpress-develop/blob/a27e920d3ebaf595f0f61c9aa66fda8f8063d318/tests/phpunit/tests/blocks/register.php#L272

```php
wp_styles()->get_data( 'unit-tests-test-block-style', 'rtl' );
```

Reply:
Brilliant :heart: 
Done in https://github.com/WordPress/gutenberg/pull/28358/commits/d222b392d64946a8f76eb11344d62f79a0c1c1bb